### PR TITLE
[BugFix]  `get_json_string` returns NULL

### LINKED ISSUE #7242: get_json_array returns NULL
### Steps to reproduce the behavior (Required)
```
mysql> SELECT get_json_string('[{"k1":1,"k2":"v2"},{"k1":2,"k2":"v2"}]', '$[*].k1');

```

### Expected behavior (Required)
| get_json_string('[{"k1":1,"k2":"v2"},{"k1":2,"k2":"v2"}]', '$[*].k1') |
+-----------------------------------------------------------------------+
| [1, 2]                                                                |
+-----------------------------------------------------------------------+
### Real behavior (Required)
```
+-----------------------------------------------------------------------+
| get_json_string('[{"k1":1,"k2":"v2"},{"k1":2,"k2":"v2"}]', '$[*].k1') |
+-----------------------------------------------------------------------+
| NULL                                                                  |
+-----------------------------------------------------------------------+

```
### StarRocks version (Required)
main, branch-2.3, branch-2.2, branch-2.1

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7242

## Problem Summary(Required) ：

This PR fixes the problem that the `get_json_string` returns NULL when a json array is the input. The previous implement of `get_json_xx` is implemented with simdjson directly, which is confusing and hard to maintain. The new implement unifies the implement of `get_json_xx` and the functions of `JsonValue` and simplifies the `get_json_xx`.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/vectorized/json_functions.h [hunk 1]
```diff
@@ -121,6 +121,20 @@ class JsonFunctions {
      */
     DEFINE_VECTORIZED_FN(json_string);
 
+    /**
+     * @param: [json_column]
+     * @paramType: [JsonColumn]
+     * @return: Int32Column
+     */
+    DEFINE_VECTORIZED_FN(json_int);
```
Comment (mofeiatwork) [inline be/src/exprs/vectorized/json_functions.h:129]: Does our user need these functions?

File: be/src/exprs/vectorized/json_functions.cpp [hunk 2]
```diff
@@ -425,6 +268,67 @@ ColumnPtr JsonFunctions::json_string(FunctionContext* context, const Columns& co
...
+                result.append_null();
+            } else {
+                result.append(std::move(json_d.value()));
+            }
+        }
+    }
+    return result.build(ColumnHelper::is_all_const(columns));
+}
+
+ColumnPtr JsonFunctions::string_json(FunctionContext* context, const Columns& columns) {
```
Comment (chaoyli) [inline be/src/exprs/vectorized/json_functions.cpp:311]: Why the other function name json_int/json_double, this function is renamed string_json other than json_string?

File: be/src/exprs/vectorized/json_functions.h (hunk 1 above)
Comment (rickif) [reply be/src/exprs/vectorized/json_functions.h]: `json_int`, `json_double` and `string_double` are all used internally, so I make them as private now.

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by murphyatwork — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**